modin/pandas/base.py

```diff
@@ -245,47 +245,44 @@ def _validate_other(
         TypeError
             If any validation checks fail.
         """
-        # We skip dtype checking if the other is a scalar.
-        if is_scalar(other):
+        if not isinstance(other, BasePandasDataset) and not is_list_like(other):
             return other
-        axis = self._get_axis_number(axis) if axis is not None else 1
-        result = other
         if isinstance(other, BasePandasDataset):
             return other._query_compiler
-        elif is_list_like(other):
-            if axis == 0:
-                if len(other) != len(self._query_compiler.index):
-                    raise ValueError(
-                        f"Unable to coerce to Series, length must be {len(self._query_compiler.index)}: "
-                        + f"given {len(other)}"
-                    )
-            else:
-                if len(other) != len(self._query_compiler.columns):
-                    raise ValueError(
-                        f"Unable to coerce to Series, length must be {len(self._query_compiler.columns)}: "
-                        + f"given {len(other)}"
-                    )
-            if hasattr(other, "dtype"):
-                other_dtypes = [other.dtype] * len(other)
-            elif is_dict_like(other):
-                other_dtypes = [
-                    type(other[label])
-                    for label in self._query_compiler.get_axis(axis)
-                    # The binary operation is applied for intersection of axis labels
-                    # and dictionary keys. So filtering out extra keys.
-                    if label in other
-                ]
-            else:
-                other_dtypes = [type(x) for x in other]
+        if not is_list_like(other):
+            # We skip dtype checking if the other is a scalar. Note that pandas
+            # is_scalar can be misleading as it is False for almost all objects,
+            # even when those objects should be treated as scalars. See e.g.
+            # https://github.com/modin-project/modin/issues/5236. Therefore, we
+            # detect scalars by checking that `other` is neither a list-like nor
+            # another BasePandasDataset.
```

Comment:
we though should keep in mind that even after this PR Modin still wouldn't support cases when a custom-integer-class is passed in a list:
```python
>>> pandas.DataFrame({"a": [1, 2], "b": [3, 4]}) + [10, CustomNumber(20)]
    a   b
0  11  23
1  12  24
```

Reply:
Good point. I added a TODO pointing to new issue https://github.com/modin-project/modin/issues/5239